scipy/stats/morestats.py

```diff
@@ -1853,29 +1854,52 @@ def bartlett(*args):
            Mathematical and Physical Sciences, Vol. 160, No.901, pp. 268-282.
 
     """
-    # Handle empty input and input that is not 1d
+
+    XX = []
     for a in args:
-        if np.asanyarray(a).size == 0:
+        if axis == 1:  # Pre-process on case of axis=1
+            a = np.asarray(a).T
```

Comment:
1D arrays stay the same after transpose. Also `XX += [a]` is sufficient Also you keep calling asarray and asanyarray. Regularize the input once and perform checks at the input. The rest should be check free.